whitesource/client.py

```diff
@@ -64,41 +64,47 @@ def save_project_tag(self, projectToken: str, key: str, value: str):
             json=body,
         )
 
-    def post_project(
+    async def ws_project(
         self,
         extra_whitesource_config: typing.Dict,
         file,
-        filename: str,
         project_name: str,
         requester_email: str,
+        length: int,
+        chunk_size: int,
+        ping_interval: int,
+        ping_timeout: int,
     ):
 
-        fields = {
-            'component': (f'{filename}.tar', file, 'application/zip'),
-            'wsConfig': json.dumps({
-                'apiKey': self.api_key,
-                'projectName': project_name,
-                'productToken': self.product_token,
-                'requesterEmail': requester_email,
-                'userKey': self.creds.user_key(),
-                'wssUrl': self.wss_endpoint,
-                'extraWsConfig': extra_whitesource_config,
-            })
+        meta_data = {
```

Comment:
a brief comment would be nice (explaining that this document will configure the interaction between client and backend).

also, I think this should be moved to the server-repository. somewhat like so (did not lookup the names):

```
import ws_api.protocol
communication_cfg = ws_api.protocol.communication_cfg(chunk_size=4096, length=<length>)
```

This way, it is a bit more obvious that this be a contract between server and client (+ the datastructure is only defined once)

Reply:
Postponed